src/CMakeLists.txt

```diff
@@ -60,8 +60,14 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 endif()
 
 find_package(Git QUIET)
-string(TIMESTAMP _timestamp "%Y%m%d")
-set(CMAKE_PROJECT_VERSION_TWEAK "${_timestamp}")
+
+if(TWEAK_VERSION)
+    set(CMAKE_PROJECT_VERSION_TWEAK "${TWEAK_VERSION}")
+else()
+    string(TIMESTAMP _timestamp "%Y%m%d")

```

Comment:
>%Y%m%d [](http://example.com/codeflow?start=33&length=6)

Why uppercase Y but lowercase m and d? Why not all the same? These are digits anyway.

Reply:
We want `"Full Year" "2-digit month" "2-digit day"` which corresponds to the format string `%Y%m%d`
See https://cmake.org/cmake/help/latest/command/string.html#timestamp for more info
```
%y
The last two digits of the current year (00-99).

%Y
The current year.

%d
The day of the current month (01-31).

%H
The hour on a 24-hour clock (00-23).

%I
The hour on a 12-hour clock (01-12).

%j
The day of the current year (001-366).

%m
The month of the current year (01-12).

%b
Abbreviated month name (e.g. Oct).

%B
Full month name (e.g. October).

%M
The minute of the current hour (00-59).
```

Reply:
Got it